code/modules/power/antimatter/shielding.dm

```diff
@@ -79,11 +84,21 @@ proc/cardinalrange(var/center)
 		for(var/obj/machinery/power/am_control_unit/AMC in cardinalrange(src))
 			if(AMC.add_shielding(src))
 				break
-		if(!priorscan)
-			sleep(20)
-			controllerscan(1)//Last chance
-			return
-		qdel(src)
+		if(!mapped) // Prevent rescanning and suicide if it's part of the map
+			if(!priorscan)
+				sleep(20)
+				controllerscan(1)//Last chance
+				return
+			qdel(src)
+
+// Find surrounding unconnected shielding and add them to our controller
+/obj/machinery/am_shielding/proc/assimilate()
```

Comment:
When the controller may not be loaded yet, it may not be possible for all the shielding to know of its existence. (For example, the Kestrel vault's AME.)
Shielding previously required either a controller, or another shielding unit _with a link to a controller_, to be directly in contact with it.
It is no longer safe to assume that these units will be in contact with anything linked to the controller.
Unless a timer is set up to continually check for a linked shield in its proximity, this recursive spreading is the only other way to associate unlinked shielding.

With regards to it being somewhat redundant code, I much prefer these 7 lines without a `sleep(20)` call to reusing `controllerscan()`